spec/lib/github_service/commands/run_test_spec.rb

```diff
@@ -0,0 +1,349 @@
+        comment_body = {
+          "body" => "@non_member Only members of the ManageIQ organization may use this command."
+        }.to_json
+        github_service_add_stub :url           => comment_url,
+                                :method        => :post,
+                                :request_body  => comment_body,
+                                :response_body => {"id" => 1234}.to_json
+
+        run_execute!(:valid => false)
+
+        github_service_stubs.verify_stubbed_calls
+      end
+    end
+
+    context "with an issue (not a PR)" do
+      before do
+        clear_stubs_for!(:get, issue_url)
+        clear_stubs_for!(:get, repo_check_url) # never reached
+
+        issue_fetch_response = single_issue_request_response(fq_repo_name, issue_id)
+        github_service_add_stub :url           => issue_url,
+                                :response_body => issue_fetch_response
+      end
+
+      it "adds a comment informing the command is being ignored by the bot" do
+        comment_body = {
+          "body" => "@NickLaMuro 'run-test(s)' command is only valid on pull requests, ignoring..."
+        }.to_json
+        github_service_add_stub :url           => comment_url,
+                                :method        => :post,
+                                :request_body  => comment_body,
+                                :response_body => {"id" => 1234}.to_json
+
+        run_execute!(:valid => false)
+
+        github_service_stubs.verify_stubbed_calls
+      end
+    end
+
+    context "with an unknown repo" do
+      let(:command_value) { "fake-repo" }
+
+      # We have determined that the bot doesn't need to validate this, and it
+      # will just fail on CI instead of trying to fail early here.
+      #
+      # We also lock down by org, so we don't need to handle for this command
+      # being abused.
+      it "it is still valid" do
+        run_execute!(:valid => true)
+
+        github_service_stubs.verify_stubbed_calls
+      end
+    end
+  end
+
+  describe "#run_tests", :with_stub_cross_repo do
+    before do
+      subject.instance_variable_set(:@issuer,     command_issuer)
+      subject.instance_variable_set(:@repos,      %w[repo1 repo2])
+      subject.instance_variable_set(:@test_repos, %w[foo bar])
+
+      clear_previous_github_service_stubs!
+
+      pull_request_data = {
+        "base"  => "master",
+        "head"  => subject.branch_name,
+        "title" => "[BOT] Cross repo test run",
+        "body"  => <<~PR_BODY
+          From Pull Request:  ManageIQ/bar#1234
+          For User:           @NickLaMuro
+        PR_BODY
+      }.to_json
+
+      github_service_add_stub :url           => "/repos/ManageIQ/cross_repo-tests-remote/pulls",
+                              :method        => :post,
+                              :request_body  => pull_request_data,
+                              :response_body => {"number" => 2345}.to_json
+
+      my_email = "NickLaMuro@loljustkidding.com"
```

Comment:
Please use `example.com` or one of the other domains reserved for documentation.